sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1098,16 +1098,24 @@ object SQLContext {
       data: Iterator[_],
       beanClass: Class[_],
       attrs: Seq[AttributeReference]): Iterator[InternalRow] = {
-    val extractors =
-      JavaTypeInference.getJavaBeanReadableProperties(beanClass).map(_.getReadMethod)
-    val methodsToConverts = extractors.zip(attrs).map { case (e, attr) =>
-      (e, CatalystTypeConverters.createToCatalystConverter(attr.dataType))
+    def createStructConverter(cls: Class[_], fieldTypes: Iterator[DataType]): Any => InternalRow = {
```

Comment:
nit: `Seq[DataType]` instead of `Iterator[DataType]`?

Reply:
I used `Iterator`s instead of `Seq`s in order to avoid creating intermediate collections. However, I agree it's more concise without that.